src/server/DeviceServer.js

```diff
@@ -327,6 +319,37 @@ class DeviceServer {
     await this._eventPublisher.publish(eventData);
   };
 
+  _onDeviceClaimCodeMessage = async (
+    message: Message,
+    device: Device,
+  ): Promise<void> => {
+    const claimCode = message.getPayload().toString();
+    const deviceID = device.getID();
+    const deviceAttributes = await this._deviceAttributeRepository.getById(deviceID);
+    if (!deviceAttributes) {
+      return;
+    }
+    if (deviceAttributes.ownerID) {
+      return;
+    }
+
+    // todo if we figure out how to delete claimCode from the device
+    // we could make the flow more clean.
+    if (deviceAttributes.claimCode !== claimCode) {
+      const claimRequestUser = await this._userRepository.getByClaimCode(claimCode);
+      if (!claimRequestUser) {
+        return;
+      }
+
+      await this._userRepository.removeClaimCode(claimRequestUser.id, claimCode);
```

Comment:
I'm thinking that maybe we don't want to remove the claim code from the user.  This is another good way to see which devices the owner has. I think it should only be removed when the user removes the claim via API.

We might be able to just remove `ownerID` from device attributes and rely on claim codes.
Doing it this way would allow us to have multiple owners on the device but I'm not sure if we want that.